addons/repair/models/account_move.py

```diff
@@ -8,6 +8,11 @@ class AccountMove(models.Model):
 
     repair_ids = fields.One2many('repair.order', 'invoice_id', readonly=True, copy=False)
 
+    def unlink(self):
+        for move in self.filtered(lambda x: x.repair_ids):
+            move.repair_ids.state = '2binvoiced'
+        return super(AccountMove, self).unlink()
```

Comment:
``` python
def unlink(self):
    self.repair_ids.state = '2binvoiced'
    return super().unlink()
```

should work as well right?

Reply:
Chances of Expected Singleton as `repair_ids` is o2m
https://github.com/odoo/odoo/blob/3353b1e385420b43dd5dfb9f156dd081237623ee/addons/repair/models/account_move.py#L9